### PR TITLE
Add imu sensor broadcaster

### PR DESCRIPTION
**Pull request by v-lopez**

### CODE REVIEW

**Review by bmagyar — CHANGES_REQUESTED** [review]

**Review by bmagyar — APPROVED** [review]

File: imu_sensor_broadcaster/doc/userdoc.rst [hunk 1]
```diff
@@ -0,0 +1,18 @@
+.. _imu_sensor_broadcaster_userdoc:
```
Comment (destogl) [inline imu_sensor_broadcaster/doc/userdoc.rst:1]: Can you link this file to the main documentation document (doc/index.rst)?

File: imu_sensor_broadcaster/src/imu_sensor_broadcaster.cpp [hunk 2]
```diff
@@ -0,0 +1,138 @@
...
+    imu_sensor_ = std::make_unique<semantic_components::IMUSensor>(
+      semantic_components::IMUSensor(sensor_name_));
+  }
+  try {
+    // register ft sensor data publisher
+    sensor_state_publisher_ = node_->create_publisher<sensor_msgs::msg::Imu>(
+      "~/imu", rclcpp::SystemDefaultsQoS());
+    realtime_publisher_ = std::make_unique<StatePublisher>(sensor_state_publisher_);
+  } catch (const std::exception & e) {
+    fprintf(
```
[the remark this answers is not in the document]
Comment (destogl) [reply imu_sensor_broadcaster/src/imu_sensor_broadcaster.cpp:71]: About that, I saw are using different methods inside `catch` statements. Sometimes `RCLCPP_ERROR` and sometimes `printf`. Let's unify this. What is the preferred method?

**Review by destogl — APPROVED** [review]
All good, just two non-mandatory comments.

Thanks!

File: imu_sensor_broadcaster/doc/userdoc.rst (hunk 1 above)
Comment (bmagyar) [reply imu_sensor_broadcaster/doc/userdoc.rst:1]: I linked it

File: imu_sensor_broadcaster/src/imu_sensor_broadcaster.cpp (hunk 2 above)
[the remark this answers is not in the document]
Comment (v-lopez) [reply imu_sensor_broadcaster/src/imu_sensor_broadcaster.cpp:71]: I think always RCLCPP_ERROR unless the error is before you intiialize the Node, in which case you have no logger. 

If that's fine I'll remove the remaining `fprintf` here.

**Review by destogl — APPROVED** [review]
I am fine with it. We just need to correct this flaky test.

File: imu_sensor_broadcaster/test/test_imu_sensor_broadcaster.cpp [hunk 3]
```diff
@@ -0,0 +1,220 @@
...
+constexpr auto NODE_SUCCESS =
+  rclcpp_lifecycle::node_interfaces::LifecycleNodeInterface::CallbackReturn::SUCCESS;
+constexpr auto NODE_ERROR =
+  rclcpp_lifecycle::node_interfaces::LifecycleNodeInterface::CallbackReturn::ERROR;
+
+rclcpp::WaitResultKind wait_for(rclcpp::SubscriptionBase::SharedPtr subscription)
+{
+  rclcpp::WaitSet wait_set;
+  wait_set.add_subscription(subscription);
+  const auto timeout = std::chrono::seconds(10);
```
Comment (destogl) [inline imu_sensor_broadcaster/test/test_imu_sensor_broadcaster.cpp:47]:
```suggestion
  const auto timeout = std::chrono::seconds(20);
```

File: imu_sensor_broadcaster/test/test_load_imu_sensor_broadcaster.cpp [hunk 4]
```diff
@@ -0,0 +1,44 @@
...
+    std::make_shared<rclcpp::executors::SingleThreadedExecutor>();
+
+  controller_manager::ControllerManager cm(
+    std::make_unique<hardware_interface::ResourceManager>(
+      ros2_control_test_assets::minimal_robot_urdf), executor, "test_controller_manager");
+
+  ASSERT_NO_THROW(
+    cm.load_controller(
+      "test_imu_sensor_broadcaster",
+      "imu_sensor_broadcaster/IMUSensorBroadcaster"));
```
Comment (destogl) [inline imu_sensor_broadcaster/test/test_load_imu_sensor_broadcaster.cpp:43]:
```suggestion
      "imu_sensor_broadcaster/IMUSensorBroadcaster"));
      
  rclcpp::shutdown();
```

**Review by bmagyar — APPROVED** [review]
Tests are sufficiently stable now

**Pull request merged**